CMakeLists.txt

```diff
@@ -391,29 +390,64 @@ if (UNIX AND NOT APPLE)
 endif ()
 
 # SIMD.
-if (USE_AVX)
-    set (USE_SSE42 TRUE)
-    set (preprocessor_definitions_common
-        ${preprocessor_definitions_common}
-        APPLESEED_USE_AVX
-    )
-endif ()
-if (USE_SSE42)
-   set (USE_SSE TRUE)
-   set (preprocessor_definitions_common
-       ${preprocessor_definitions_common}
-       APPLESEED_USE_SSE42
-   )
-endif ()
-if (USE_SSE)
-   if (CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_SIZEOF_VOID_P MATCHES 4)
-       message (WARNING "Building appleseed with SSE/SSE2 instruction sets on 32-bit Linux is not supported; continuing without SSE/SSE2.")
-   else ()
+if (is_x86)
+    if (USE_F16C)
+        set (preprocessor_definitions_common
+            ${preprocessor_definitions_common}
+            APPLESEED_USE_F16C
+        )
+    endif ()
+    if (USE_FMA)
+        set (preprocessor_definitions_common
+            ${preprocessor_definitions_common}
+            APPLESEED_USE_FMA
+        )
+    endif ()
+    if (USE_AVX2)
+        set (USE_AVX TRUE)
+        set (preprocessor_definitions_common
+            ${preprocessor_definitions_common}
+            APPLESEED_USE_AVX2
+        )
+    endif ()
+    if (USE_AVX)
+        set (USE_SSE42 TRUE)
+        set (preprocessor_definitions_common
+            ${preprocessor_definitions_common}
+            APPLESEED_USE_AVX
+        )
+    endif ()
+    if (USE_SSE42)
+       set (USE_SSE TRUE)
        set (preprocessor_definitions_common
            ${preprocessor_definitions_common}
-           APPLESEED_USE_SSE
+           APPLESEED_USE_SSE42
        )
-   endif ()
+    endif ()
+    if (USE_SSE)
+       if (CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_SIZEOF_VOID_P MATCHES 4)
+           message (WARNING "Building appleseed with SSE/SSE2 instruction sets on 32-bit Linux is not supported; continuing without SSE/SSE2.")
```

Comment:
I'm not sure I understand what is the problem. I didn't do any change here.
I think that if you are on a 32 bits intel system, the only option you have is SSE.
The others only show on 64 bits systems.